PSD-2481 - Test report and Risk assessment hyperlinks

Nashar101 opened the pull request:
This Branch addresses the Jira ticket PSD-2481: https://regulatorydelivery.atlassian.net/jira/software/c/projects/PSD/boards/57?selectedIssue=PSD-2481

Description
I have changed the text adjacent to Risk assessments in the Notification page to a hyperlink that when clicked would redirect the user straight to the show page of that risk assessment

Screen-shots or screen-capture of UI changes

![image](https://github.com/OfficeForProductSafetyAndStandards/product-safety-database/assets/83423212/b02816d0-0b56-4e16-93bc-14a6d21ae7eb)

![image](https://github.com/OfficeForProductSafetyAndStandards/product-safety-database/assets/83423212/5869109b-db11-4cd6-8fc0-ffcb4af4cd4c)

alan-at-work reviewed (commented):
Interesting way of approaching this. Do we have test coverage on this to make sure the links are being built correctly?

rubenarakelyan commented:
Why not just change what `formatted_risk_assessments` does to include links? That method is only used in two places, both of which are part of the notification task list workflow. The method is a helper and should pass a complete string to the view to display, so that you don't have to construct a string inside the view itself.

Nashar101 commented:
> Why not just change what `formatted_risk_assessments` does to include links? That method is only used in two places, both of which are part of the notification task list workflow. The method is a helper and should pass a complete string to the view to display, so that you don't have to construct a string inside the view itself.

I have removed the hyperlink function and added the changes to the 'formatted_risk_assessments' function

Nashar101 commented:
> Interesting way of approaching this. Do we have test coverage on this to make sure the links are being built correctly?

Created Rspec for new feature

alan-at-work reviewed (approved):
Nice work

rubenarakelyan commented on app/views/notifications/show.html.erb:
```diff
@@ -153,7 +153,7 @@
             {
               key: { text: "Address" },
-              value: { text: formatted_business_address(investigation_business.business.locations.find_by(name: "Registered office address")).html_safe.presence || "Not provided" }
+              value: { text: "Not provided" || "Not provided" }
```
Is this intentional?

alan-at-work replied:
I did wonder about this, and why it was duplicated.

Nashar101 replied:
Unintentional mistake will revert

rubenarakelyan commented on app/views/notifications/show.html.erb:
```diff
@@ -166,6 +166,14 @@
+      <%= risk_assessment.first.id %>
+      <% risk_text = "" %>
+      <% if risk_assessment.nil? %>
+        <% risk_text = "Not provided" %>
```
Can this be done in the `formatted_risk_assessments` method rather than the view?

Nashar101 replied:
> Can this be done in the `formatted_risk_assessments` method rather than the view?

I have added the requested changes, the conditional statement is now part of the helper method
